drivers/flash/nrf_qspi_nor.c

```diff
@@ -1142,6 +1202,40 @@ static int qspi_nor_pm_action(const struct device *dev,
 }
 #endif /* CONFIG_PM_DEVICE */
 
+void  z_impl_nrf_qspi_nor_base_clock_div_force(const struct device *dev,
+					       bool force)
+{
+#if defined(BASE_CLOCK_DIV) && defined(CONFIG_SOC_SERIES_NRF53X)
+	struct qspi_nor_data *dev_data = dev->data;
+	/*
+	 * The divider is normally changed, unless the flag is set, only for
+	 * periods when the driver is locked, so the flag itself also can only
+	 * be modified while the driver is locked.
+	 */
+	qspi_lock(dev);
+	dev_data->keep_base_clock_div_set = force;
```

Comment:
On a "unforce" call the clock divider won't be restored to NRF_CLOCK_HFCLK_DIV_4. So the divider might have unexpected value unless next QSPI operation. Similar problem might affects a "force" call.
I don't know whether I'm exaggerating with doubts.

Reply:
> On a "unforce" call the clock divider won't be restored to NRF_CLOCK_HFCLK_DIV_4.

It will be restored - `dev_data->keep_base_clock_div_set` is set here to `false` and then `qspi_unlock()` is called, where the following code is executed:
```
	if (!dev_data->keep_base_clock_div_set) {
		nrf_clock_hfclk192m_div_set(NRF_CLOCK, NRF_CLOCK_HFCLK_DIV_4);
	}
```
Similarly for the "force" call - the divider is changed in `qspi_lock()` (as `dev_data->keep_base_clock_div_set` is `false`), then `dev_data->keep_base_clock_div_set` is set here to `true`, what prevents the divider from being restored to default in `qspi_unlock()` that is called next.

Reply:
Of course you are right.